### ISSUE TITLE
ELM327 unable to connect to car

### ISSUE THREAD

**Issue by stibbzy** [meatpiHQ/wican-fw#144]
I just received my WiCan (USB) and having trouble on my BMW i3. When I use  Carscanner with a BT OBD2 adaptor it connects just fine and I can get sensor data from the car. 
However, when I connect Carscanner to the WiCan, it says “succesfully connected to ELM327, but ELM327 wasn't able to establish connection with your car.”
The green LED is on and the orange LED is flashing. 
I’ve tried turning MQTT on and off, and disabled battery alerts. Different bitrates don’t work either. The Carscanner log has a lot of “BUS INIT: …ERROR” messages.
I've attached screenshots from Carscanner when it is working via the BT adaptor and when it isn't via WiCan. I've attached the log as well.

![IMG_2885](https://github.com/meatpiHQ/wican-fw/assets/55930421/6be23471-4e1f-48ea-b17c-0e32bbdf75c4)
![IMG_2884](https://github.com/meatpiHQ/wican-fw/assets/55930421/e19847ac-759e-48e2-962a-da172a40aa53)
![IMG_2883](https://github.com/meatpiHQ/wican-fw/assets/55930421/198047e3-bf87-494b-947e-cfc9b55b72a0)
![IMG_2881](https://github.com/meatpiHQ/wican-fw/assets/55930421/43010bbd-3bb6-4413-88b8-69af71cd01fb)
![IMG_2879](https://github.com/meatpiHQ/wican-fw/assets/55930421/52062c15-3ab3-46dc-b9a5-34505b5b29da)
![IMG_2878](https://github.com/meatpiHQ/wican-fw/assets/55930421/813ee2e0-38f0-4bdd-a64e-b8497c68600b)
![IMG_2877](https://github.com/meatpiHQ/wican-fw/assets/55930421/d714dc05-f1f9-4a02-beff-1b8a966e19f1)
![IMG_2876](https://github.com/meatpiHQ/wican-fw/assets/55930421/a66836c9-767f-4385-938d-5902adbc68f0)
![IMG_2875](https://github.com/meatpiHQ/wican-fw/assets/55930421/ce3e26b0-659d-40c1-9cc2-91f09cf25973)

[Carscanner log.txt](https://github.com/user-attachments/files/15921317/Carscanner.log.txt)

**Comment by harkoprinssen** [comment]
Last screenshot says it's connected to the ECU. Still not able to read data?

**Comment by stibbzy** [comment]
> Last screenshot says it's connected to the ECU. Still not able to read data?

No, that is a screenshot illustrating how the Bluetooth OBD2 adaptor connects just fine to the ECU. Carscanner cannot connect to the ECU when using WiCAN

**Comment by harkoprinssen** [comment]
Ah, now I see.
Behaviour seems like the issue I am facing, however I do not get the orange light blinking connected to my car

**Comment by stibbzy** [comment]
> @stibbzy do you have both adapters connected to the same bus at the same time?

No, one at a time. I turn off the car in between.

**Comment by meatpiHQ** [comment]
@stibbzy Try setting it to AP mode, disable mqtt logging and MQTT. Submit the settings and do not use the Monitor tab. The. Then try to connect.

**Comment by meatpiHQ** [comment]
@stibbzy looking at the logs... seems that there are some elm327 commands that are not supported by WiCAN. I will try to add them and upload new firmware version for you to try.

**Comment by harkoprinssen** [comment]
Could that possibly also help me out? https://github.com/meatpiHQ/wican-fw/issues/142

> @stibbzy looking at the logs... seems that there are some elm327 commands that are not supported by WiCAN. I will try to add them and upload new firmware version for you to try.

**Comment by stibbzy** [comment]
@meatpiHQ  Thanks so much!
I’ll also try your other suggestion in the morning (Sydney time)

**Comment by meatpiHQ** [comment]
@harkoprinssen I'm looking at your issue now.

**Comment by stibbzy** [comment]
@meatpiHQ I just tried setting to AP mode, disabled MQTT and MQTT logging, and didn't use the Monitor tab. Connected to the WiCAN's wifi and ran Carscanner, same result - can't connect to ECU.
I also set Carscanner to run through 17 different CAN protocols to try to detect the right one (different bit count, baud rate etc) and none of those worked.

**Comment by stibbzy** [comment]
@meatpiHQ I just spent another hour in the car trying different things but nothing worked. For what it's worth, Carscanner has a toggle to detect fake ELM327 clones. When I toggle that on, it detects the WiCAN as fake... not sure if that means anything or not. 
Also, under Carscanner advance settings, it can display AT commands that aren't supported by the adaptor. When I use the BT adapter, that field is blank, but with the WiCAN, it displays all these.

![IMG_2892](https://github.com/meatpiHQ/wican-fw/assets/55930421/cad5b803-bf3c-4546-83a9-479b73b97ba0)
![IMG_2894](https://github.com/meatpiHQ/wican-fw/assets/55930421/b9b09d1d-cc9c-44f6-b2f9-a71008969bee)

**Comment by meatpiHQ** [comment]
@stibbzy WiCAN only emulates an ELM327 device and is not intended for diagnostics. It allows for extracting and logging data from the bus for reverse engineering and integration with Home Assistant. While this has worked with most cars, some vehicles require specific commands to function properly.

One approach is to log data from your other BLE adapter and use this information to configure WiCAN.

**Comment by stibbzy** [comment]
@meatpiHQ Yep, all I want to be able to do is to pull the battery SOC into Home Assistant. Would an updated firmware likely start giving responses in CarScanner?

**Comment by meatpiHQ** [comment]
> Would an updated firmware likely start giving responses in CarScanner?

Yes, that's the plan. But it might be faster if you log the SoC data using the other EM327 adapter, by clicking on the sensor on carscanner. All we need is the PID request and response.

**Comment by stibbzy** [comment]
Ok, here is the log once I clicked the sensor and let it go for about 10 seconds...
[Uploading BMW i3 Batteru SOC.txt…]()

**Comment by meatpiHQ** [comment]
22DDBC is the OBD request.

These frames are the PID request and response. We need to translate these into CAN frames and send it over MQTT. I'll have a look at it later tonight.

**Comment by stibbzy** [comment]
Thanks so much!

**Comment by meatpiHQ** [comment]
I think the PID request should be:

> {"payload":{"bus":"0","type":"tx","ts":0,"frame":[{"id":1777,"dlc":8,"rtr":false,"extd":true,"data":[3,34,221,188,170,170,170,170]}]}}

After you send this frame you should expect a rx frame with the first data byte is 16, once you receive that, you have to send a response frame:

> {"payload":{"bus":"0","type":"tx","ts":0,"frame":[{"id":1777,"dlc":8,"rtr":false,"extd":true,"data":[48,0,10,170,170,170,170,170]}]}}

This issue is a good reference. 

https://github.com/meatpiHQ/wican-fw/issues/97#issuecomment-2089644589

**Comment by stibbzy** [comment]
I'm still not understanding... Do I publish that to the topic directly under wican (wican/ecda3ba6d52d)? I tried, but I didn't get any rx response. That's what I don't understand, I have never seen a rx response in MQTT Explorer:
<img width="1492" alt="Screenshot 2024-06-23 at 7 48 33 AM" src="https://github.com/meatpiHQ/wican-fw/assets/55930421/29b8b095-32e5-44f6-9d28-4d86411fe23a">

**Comment by harkoprinssen** [comment]
I think you should be publishing in the ../tx topic. 

Mine goes to wican/f412fa377d01/can/tx

**Comment by stibbzy** [comment]
@harkoprinssen Ahh, thanks very much for the suggestion. Unfortunately it didn't give a response on either wican/ecda3ba6d52d/can/tx or wican/ecda3ba6d52d/elm327/tx

**Comment by meatpiHQ** [comment]
Yes to the tx topic. Make sure you have " MQTT ELM327 logging" disabled.

**Comment by stibbzy** [comment]
@meatpiHQ OMG, I finally have a response after disabling logging!
{"bus":"0","type":"rx","ts":37927,"frame":[{"id":304,"dlc":5,"rtr":false,"extd":false,"data":[5,240,252,255,255,0,0,0]}]}

**Comment by meatpiHQ** [comment]
That's great, but it doesn't look like the frame we're after.

Btw you can have the BLE and WiCAN adapter connected at the same time if you want, you can see Rx frames over MQTT.

**Comment by stibbzy** [comment]
I'm using the wiCAN adaptor with a spliced OBD2 cable at the moment, so the BLE can't be plugged in as well unfortunately. Is there only supposed to be one frame with the 16 byte? If so, I would have missed it with all the messages coming in

**Comment by meatpiHQ** [comment]
> If so, I would have missed it with all the messages coming in. All the messages are identical, apart from the ts value of course.

what are the messages coming in? 

Note: You should only add the filter after we figured out the message structure, if you added a filter you won't see a more messages coming only the filtered ID.

**Comment by stibbzy** [comment]
@meatpiHQ Without that filter, there are ~1000 messages, and they all say {"bus":"0","type":"rx","ts":37927,"frame":[{"id":304,"dlc":5,"rtr":false,"extd":false,"data":[5,240,252,255,255,0,0,0]}]}

I was just trying the filter to try to see if there was my desired response buried in there. But I don't know how to determine the CAN ID number or the PID that the reply would come back on. Based on this BLE response from before, I just tried CAN ID 1543 (Hex 607) and PID 241 (Hex F1) but that didn't work either.

**Comment by meatpiHQ** [comment]
Since your car broadcasts over the CAN bus we can try a different approach.

Try adding this filter: 

ID: 1074
Name: SoC
PID: -1
index: 0
Start Bit: 0
Bit Length: 1
Expression:  B4/2
Cycle ms: 1000

I found a DBC file for I3 not sure if its compatible with your car model, but you should give it a go.

https://github.com/damienmaguire/BMW-i3-CCS/blob/main/i3_LIM_dbc1.dbc

**Comment by stibbzy** [comment]
@meatpiHQ Thanks for finding the DBC file....  it would very likely be compatible, but I have no idea what a DBC file is or how to use it.

I tried your filter, then published the tx json, but there was nothing received. Are my screenshots correct?

<img width="734" alt="Screenshot 2024-06-23 at 12 00 54 PM" src="https://github.com/meatpiHQ/wican-fw/assets/55930421/abdf421f-8b18-4d41-823d-a02e34cb092d">
<img width="1486" alt="Screenshot 2024-06-23 at 12 01 35 PM" src="https://github.com/meatpiHQ/wican-fw/assets/55930421/d642a492-e344-4a8f-aa95-c3a2c08945fb">

**Comment by meatpiHQ** [comment]
Since your car is always broadcasting then there is no need to publish any tx frame. Did you store and submit ? If it works you should see an mqtt message like : {"SoC": 51}


The DBC file is a database file that would tell you have to interpret CAN frames:

The CAN frame id is 1074. The Display_SoC is what ever your car display is showing, 32|8@1+ means that the SoC value is in Byte 4 
<img width="413" alt="image" src="https://github.com/meatpiHQ/wican-fw/assets/94690098/9ac7013c-28fc-4054-8aac-dcf3ae8d1df4">

**Comment by stibbzy** [comment]
@meatpiHQ Yes, I stored and submitted, and after refreshing browser, the filter was there.
The rx topic is not receiving any messages. I guess my question was, with no filter, why is every received response identical?

**Comment by meatpiHQ** [comment]
You mean that the car is always broadcasting the same frame? is the engine ON?

**Comment by stibbzy** [comment]
@meatpiHQ There is no engine as such (electric), but it is powered on, in drive, with my foot on the brake. Without a filter, every single frame (1000s per second) is:  {"bus":"0","type":"rx","ts":37927,"frame":[{"id":304,"dlc":5,"rtr":false,"extd":false,"data":[5,240,252,255,255,0,0,0]}]}

**Comment by meatpiHQ** [comment]
Ok. Motor then :) 

This frame is the like a keep alive frame, not useful for what your trying to do. This tells us that the DBC file is not compatible.

**Comment by stibbzy** [comment]
@meatpiHQ Ok, so I probably do need a TX frame, and then somewhere within the keep alive frames will be my response frame, once we figure out the frame structure?

Is there any way to filter out the keep alive messages lol?

**Comment by meatpiHQ** [comment]
> so I probably do need a TX frame, and then somewhere within the keep alive frames will be my response frame, once we figure out the frame structure?

Yes you need to send the TX frame.

Try this firmware, it will publish the CAN frame by ID. So each frame with different CAN ID will go to a different topic.

wican/ecda3ba6d52d/can/rx/xxxxx

(Extract the bin file from the zip and upload it to the device)
[wican-fw_v299_hv300.zip](https://github.com/user-attachments/files/15942546/wican-fw_v299_hv300.zip)

**Comment by stibbzy** [comment]
@meatpiHQ OK, I'll give it a go.

**Comment by stibbzy** [comment]
@meatpiHQ 304 is the only frame, unless I push a tx frame...
![Uploading Screenshot 2024-06-23 at 3.01.33 PM.png…]()

**Comment by meatpiHQ** [comment]
What are the other IDs if you push the tx frame?

No the carscanner is not working because sending an unsupported command, that changes the message format so it's failing to parse the response.

**Comment by stibbzy** [comment]
@meatpiHQ I don't get any other IDs if I push a TX frame. In my screenshot, I show the response with a TX frame, it just has 304 rx, plus the transmitted frame

**Comment by meatpiHQ** [comment]
What about if you send this frame?

{ bus : 0 , type : tx , frame :[{ id :2015, dlc :8, rtr :false, extd :false, data :[2,1,0,170,170,170,170,170]}]}

**Comment by stibbzy** [comment]
@meatpiHQ No change:
<img width="1242" alt="Screenshot 2024-06-23 at 3 16 10 PM" src="https://github.com/meatpiHQ/wican-fw/assets/55930421/6b0de47d-a37d-42fa-98db-414635673a0c">

**Comment by meatpiHQ** [comment]
seems like its a problem with the mqtt explore, try this

{
   "bus":0,
   "type":"tx",
   "frame":[
      {
         "id":2015,
         "dlc":8,
         "rtr":false,
         "extd":false,
         "data":[
            2,
            1,
            0,
            170,
            170,
            170,
            170,
            170
         ]
      }
   ]
}

**Comment by stibbzy** [comment]
@meatpiHQ No change, I've also tried issuing these from the Home Assistant MQTT add on with no luck.
<img width="1242" alt="Screenshot 2024-06-23 at 3 28 45 PM" src="https://github.com/meatpiHQ/wican-fw/assets/55930421/3e409b54-07a3-434a-baf4-3369dc196f4c">

**Comment by meatpiHQ** [comment]
Something does not add up.. Just to confirm please double check the tx topic in the configuration page?

**Comment by stibbzy** [comment]
It matches:
<img width="1242" alt="Screenshot 2024-06-23 at 3 35 04 PM" src="https://github.com/meatpiHQ/wican-fw/assets/55930421/ba06c6c7-7b1a-4aea-a19a-4c5407b6872a">

**Comment by meatpiHQ** [comment]
try this frame again in this format

{
   "payload":{
      "bus":"0",
      "type":"tx",
      "ts":0,
      "frame":[
         {
            "id":1777,
            "dlc":8,
            "rtr":false,
            "extd":true,
            "data":[
               3,
               34,
               221,
               188,
               170,
               170,
               170,
               170
            ]
         }
      ]
   }
}

**Comment by stibbzy** [comment]
@meatpiHQ Nope... are we sure these are real frames that are being received? 
<img width="1512" alt="Screenshot 2024-06-23 at 3 40 46 PM" src="https://github.com/meatpiHQ/wican-fw/assets/55930421/0963120c-60ff-417f-aaf3-eb32e5a9f8d0">

**Comment by meatpiHQ** [comment]
> { "payload":{ "bus":"0", "type":"tx", "ts":0, "frame":[ { "id":1777, "dlc":8, "rtr":false, "extd":true, "data":[ 3, 34, 221, 188, 170, 170, 170, 170 ] } ] } }

I accidently had the extended flag turned ON "extd":true , try the frame below. 



{
"payload":{
"bus":"0",
"type":"tx",
"ts":0,
"frame":[
{
"id":1777,
"dlc":8,
"rtr":false,
"extd":false,
"data":[
3,
34,
221,
188,
170,
170,
170,
170
]
}
]
}
}




> are we sure these are real frames that are being received?

Yes, I'm just translating them into CAN frames.

**Comment by stibbzy** [comment]
@meatpiHQ That didn't have any effect, however, I had left the car just sitting with RX frames pouring in, and a few other IDs showed up, but at much less frequency than the 304 ID:
<img width="1512" alt="Screenshot 2024-06-23 at 4 17 10 PM" src="https://github.com/meatpiHQ/wican-fw/assets/55930421/275cb493-4444-4b11-ab6d-47f3628a0589">

**Comment by meatpiHQ** [comment]
can you show me what's in the frames? part of them are hidden in the screenshot

**Comment by stibbzy** [comment]
@meatpiHQ This is for 1815:
{"bus":"0","type":"rx","ts":35227,"frame":[{"id":126,"dlc":15,"rtr":true,"extd":false,"data":[0,0,0,0,0,0,0,0]},{"id":75126616,"dlc":10,"rtr":true,"extd":true,"data":[0,0,0,0,0,0,0,0]},{"id":160979277,"dlc":3,"rtr":false,"extd":true,"data":[99,6,33,0,0,0,0,0]},{"id":443501415,"dlc":2,"rtr":true,"extd":true,"data":[0,0,0,0,0,0,0,0]},{"id":1815,"dlc":1,"rtr":true,"extd":false,"data":[0,0,0,0,0,0,0,0]}]}

For 443501415:
{"bus":"0","type":"rx","ts":37716,"frame":[{"id":286,"dlc":10,"rtr":true,"extd":false,"data":[0,0,0,0,0,0,0,0]},{"id":607,"dlc":15,"rtr":true,"extd":false,"data":[0,0,0,0,0,0,0,0]},{"id":161241413,"dlc":3,"rtr":true,"extd":true,"data":[0,0,0,0,0,0,0,0]},{"id":1441,"dlc":3,"rtr":false,"extd":false,"data":[37,205,16,0,0,0,0,0]},{"id":443501415,"dlc":2,"rtr":true,"extd":true,"data":[0,0,0,0,0,0,0,0]}]}

For 1967:
{"bus":"0","type":"rx","ts":37718,"frame":[{"id":475920656,"dlc":3,"rtr":true,"extd":true,"data":[0,0,0,0,0,0,0,0]},{"id":1967,"dlc":12,"rtr":true,"extd":false,"data":[0,0,0,0,0,0,0,0]}]}

**Comment by meatpiHQ** [comment]
I've never seen anything like this before... dlc= 15 ?? how is that even possible. DLC in a CAN frame is max 8.

**Comment by stibbzy** [comment]
@meatpiHQ Oh dear lol.
All this was brought about because the 3g network in Australia was shut down this month, and now a group of us on Facebook are trying to find a solution to get at least the SOC out of the car.

**Comment by meatpiHQ** [comment]
ah so the i3 has a 3G connection?  

I can send you the new WiCAN pro adapter, it supports all the possible OBD commands. And will let you connect to the LTE using an accessory. https://www.crowdsupply.com/meatpi-electronics/wican-pro

**Comment by stibbzy** [comment]
@meatpiHQ  Oh really? Should that work out of the box like carscanner then if it supports all commands?

**Comment by meatpiHQ** [comment]
yes absolutely, that's what it's supposed to do. It's WiCAN on steroids or more like an OBDLINK MX with an esp32 as an application processor. The firmware is not 100% yet but I will have something soon.

Send me an email with your shipping address: https://www.meatpi.com/contact

> It is working with OVMS... is there anything in the GitHub code for that which would help?

I saw the code it's massive, it will take me a while to find it.

**Comment by stibbzy** [comment]
@meatpiHQ OK, will do!
This is the i3 portion of the OVMS anyway, for what it's worth:
https://github.com/openvehicles/Open-Vehicle-Monitoring-System-3/tree/master/vehicle/OVMS.V3/components/vehicle_bmwi3

**Comment by meatpiHQ** [comment]
Hey @stibbzy please try this on WiCAN Pro. Extract the firmware from the zip file and update the device. 

[wican-fw_obd_pro_v298-5-gc28f6be-dirty.zip](https://github.com/user-attachments/files/16119291/wican-fw_obd_pro_v298-5-gc28f6be-dirty.zip) 

First set the protocol to: AutoPID

<img width="251" alt="image" src="https://github.com/meatpiHQ/wican-fw/assets/94690098/202be11b-fb34-45e6-9284-551851ab97b7">

Then add a new custom pid and fill in the parameters below, then press the **store** button then reboot the device. If it works you should see a mqtt message on topic "soc_topic" with the soc value.

**Initialisation**: ATAT1;ATCEA07;ATTAF1;ATFCSH6F1;ATSH6F1;ATFCSD07300000;ATFCSM1;ATCRA607;
**Name**:  SoC
**PID**: 22DDBC
**Expression**: [B6:B7]/10
**Period**: 2000
**Send_to**: soc_topic


<img width="836" alt="image" src="https://github.com/meatpiHQ/wican-fw/assets/94690098/753fd9bc-3415-4d1f-95e3-1829f7e65917">

**Comment by stibbzy** [comment]
@meatpiHQ Ahh so close!  So after configuring as above, the soc_topic is returning 410.5... My battery SoC reported by Carscanner (when in elm327 mode) is 46.4%...
I tried with 'Expression' set at just 'V' but it returned 0.

**Comment by meatpiHQ** [comment]
@stibbzy can you please send me the carscanner log when using WiCAN pro to read the SoC? Set the protocol back to elm327 to carscanner.

**Comment by stibbzy** [comment]
@meatpiHQ Log is attached, with tonight’s results right down the bottom. 
[Bmwi3log08072024.txt](https://github.com/user-attachments/files/16125421/Bmwi3log08072024.txt)

**Comment by meatpiHQ** [comment]
@stibbzy Please try this firmware with this new Initialisation string, I also added "raw" value to the mqtt message, this will help to debug the issue. 

**Initialisation:**  

ATE0;ATH1;ATAL;ATPBE101;ATSPB;ATBI;ATSH6F1;ATAT0;ATSTFF;ATCRA607;ATCEA07;ATFCSH6F1;ATFCSD07300800;ATFCSM1;

[wican-fw_obd_pro_v298-7-g92a9d58-dirty.zip](https://github.com/user-attachments/files/16131448/wican-fw_obd_pro_v298-7-g92a9d58-dirty.zip)

> I tried with 'Expression' set at just 'V' but it returned 0.

You cannot use V, instead I enabled byte array. [B6:B7]/10 will take the value of byte 6 and 7 and divide it by 10.

**Comment by stibbzy** [comment]
@meatpiHQ Ok, so after applying the firmware, the calculated result is the same (410.5), but I can see the raw frame and can  see it reporting the correct SoC. This is a screenshot:
<img width="728" alt="Screenshot 2024-07-09 at 8 11 29 AM" src="https://github.com/meatpiHQ/wican-fw/assets/55930421/4c1309c1-2d33-4966-8de7-4d41ffae656b">

Per my previous comment:

> 
> Apparently, 22DDBC is the battery SOC and that is present in my log, eg:
> 
> > 22DDBC
> > 607F1037F2278
> > 607F1100962DDBC027D
> > 607F121031C0050FFFF
> 
> If I just take the first frame and convert the last byte? 027D into decimal, I get 637, and my current SOC according to Carscanner is 63.7%. I then ran the car down to 63% SOC according to Carscanner, and the byte returned was 0276 in the log, which is a perfect match.

I can see that the value after 62DDBC in my new result is 03E8, which is decimal 1000. Divide by 10 is 100. Since I fully changed the car last night, this is a match for the actual SoC.

**Comment by meatpiHQ** [comment]
@stibbzy ok I know whats going on. Try this

Expression: [B11:B12]/10

**Comment by stibbzy** [comment]
@meatpiHQ That gave 844.8, which is 2100 hex, which is in the second frame.
I played around with the byte array while referencing the response, and it really seemed like [B11:B12]/10 was correct.
I finally set it back to that, and suddenly it reported SoC 100! But the raw response was 0A (10 dec).
Pulled it out of the OBD port and reinserted, the SoC remained at 100 and the full frame was reported again.

So it seems to be working, apart from the battery voltage!

**Comment by meatpiHQ** [comment]
@stibbzy Great! I still don't fully understand this, the response for the PID request seems to be either 2 frames or 3 frames. BMW are always so special. Let me know if you see any jump in the battery SoC value. 

> So it seems to be working, apart from the battery voltage!

I'm aware of the issue, this still needs a bit of work.

Btw have you tried BimmerCode and BimmerLink Apps?

**Comment by stibbzy** [comment]
@meatpiHQ Yep, I've used BimmerCode to disable the OBD alarm and modify other features.
So if I leave the unit plugged in, will it detect the 12v level and go into low power mode at the moment?

**Comment by meatpiHQ** [comment]
@stibbzy good to know that it works with BimmerCode.

> So if I leave the unit plugged in, will it detect the 12v level and go into low power mode at the moment?

Lower power mode is not ready yet, still needs a bit more work. I'll send you a firmware update when it's ready.

**Comment by stibbzy** [comment]
@meatpiHQ Ahh ok, that means it's not usable as yet as a permanent install.

Can the WiCAN Pro be powered externally? That is why I originally went with the WiCAN USB, as I have a nice 70Wh battery in the car powering my Dashcam.

**Comment by meatpiHQ** [comment]
@stibbzy yeah not ready for permanent install yet but it will be soon.

No it cannot be powered from the USB connector.

**Comment by zitnyp** [comment]
> ah so the i3 has a 3G connection?
> 
> I can send you the new WiCAN pro adapter, it supports all the possible OBD commands. And will let you connect to the LTE using an accessory. https://www.crowdsupply.com/meatpi-electronics/wican-pro

Hello,

did I understand correctly that WiCAN (ESP32-C3) does not work with BMW i3 SOC batteries?

can/bmwi3/automate/error {"error": "Timeout, pid: 62DDBC"}

![image](https://github.com/user-attachments/assets/6535bae2-f5e9-4606-a6fb-9ecdfb840f40)

**Comment by meatpiHQ** [comment]
@zitnyp That's right, we could not get it to work on the I3. 

It works fine on the PRO version, but the hardware and firmware are not yet released.

**Comment by zitnyp** [comment]
@meatpiHQ and is it a HW problem or a SW/Firmware problem?

I have the i3rex available and I can get the temperature of the coolant just fine.

**Comment by meatpiHQ** [comment]
@zitnyp How are you reading it? are you using car scanner? if so it would be great if you can export the logs.

It's most likely an initialization issue.

**Comment by lostOzone** [comment]
Is it working with the WiCan now or still the pro? I ordered after reading the supported vehicules list but there was no mention about the pro version being required.
